services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -491,15 +491,57 @@ async def try_update_database_from_storage(
     )
     async def auto_update_database_from_storage_task(
         self, file_uuid: str, bucket_name: str, object_name: str
-    ):
+    ) -> Optional[FileMetaDataEx]:
         return await self.try_update_database_from_storage(
             file_uuid, bucket_name, object_name, silence_exception=True
         )
 
-    async def upload_link(self, user_id: str, file_uuid: str, as_presigned_link: bool):
+    async def update_metadata(
+        self, file_uuid: str, user_id: int
+    ) -> Optional[FileMetaDataEx]:
+        async with self.engine.acquire() as conn:
+            can: Optional[AccessRights] = await get_file_access_rights(
+                conn, int(user_id), file_uuid
+            )
+            if not can.write:
+                message = f"User {user_id} was not allowed to upload file {file_uuid}"
+                logger.debug(message)
+                raise web.HTTPForbidden(reason=message)
+
+        bucket_name = self.simcore_bucket_name
+        object_name = file_uuid
+        return await self.auto_update_database_from_storage_task(
+            file_uuid=file_uuid,
+            bucket_name=bucket_name,
+            object_name=object_name,
+        )
+
+    async def delete_metadata(self, user_id: int, file_uuid: str) -> None:
+        async with self.engine.acquire() as conn:
+            can: Optional[AccessRights] = await get_file_access_rights(
+                conn, int(user_id), file_uuid
+            )
+            if not can.write:
+                message = f"User {user_id} was not allowed to upload file {file_uuid}"
+                logger.debug(message)
+                raise web.HTTPForbidden(reason=message)
+
+            try:
+                await conn.execute(
+                    file_meta_data.delete().where(
+                        file_meta_data.c.file_uuid == file_uuid
+                    )
+                )
+            except Exception:
```

Comment:
As I mentioned in last review, PLEASE do not use these generic exceptions ... in addition translating to not-found is very specific. 
Make sure the translation from db errors to HTTP errors is coherent !! (e.g. ``Exception``-> 500 etc)

To see db errors look at https://github.com/ITISFoundation/osparc-simcore/blob/5ef4cd985f98f1ca4ee116659624748c5bf683a8/packages/postgres-database/src/simcore_postgres_database/errors.py

Reply:
Good point will change and let the `upload_file` function call the metadata update.